### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by tezc**

### CODE REVIEW

File: docs/Using.md [hunk 1]
```diff
@@ -0,0 +1,461 @@
+Using the library
```
Comment (tezc) [inline docs/Using.md:1]: I guess we should start with TOC. Open to suggestions. 
Listing topics that comes to my mind:

1- A short section about implementing callbacks 
2- Initializing library
3- Initializing cluster (first time use)
4- Submit requests
5- Log compaction
6- What to do after a restart
7- Sending and receiving a snapshot
....

8- Adding node and removing nodes? 
9- What else?

File: docs/Using.md [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+
+
+### Libraft callbacks
+You provide your callbacks to the Raft server using `raft_set_callbacks()`.
+Application must implement all the mandatory callbacks. You can find more detailed info in the `raft.h`.
+
+```c
+typedef struct
+{
+    raft_logentry_event_f applylog;
```
Comment (tezc) [inline docs/Using.md:17]: We should have a few lines of explanation of callbacks (also callbacks without typedef)

File: docs/Using.md (hunk 1 above)
Comment (tezc) [reply docs/Using.md:1]: Shaya: Can we provide disk based log implementation example?

**Review by fadidahanna — APPROVED** [review]
Great documentation, @tezc!

**Pull request merged**